plugins/JetConstituentTableProducer.cc

```diff
@@ -256,22 +321,47 @@ void JetConstituentTableProducer<T>::produce(edm::Event &iEvent, const edm::Even
   iEvent.put(std::move(svTable), nameSV_);
 
   iEvent.put(std::move(outCands));
+
+
+	// Muon table
+	auto muonTable = std::make_unique<nanoaod::FlatTable>(outMuons->size(), nameMu_, false);
+	// We fill from here only stuff that cannot be created with the SimpleFlatTnameableProducer
+	muonTable->addColumn<int>("jetIdx", jetIdx_mu, "Index of the parent jet");
+	muonTable->addColumn<int>(idx_nameMu_, muIdx, "Index in the Muon list");
+	if (readBtag_) {
+		muonTable->addColumn<float>("pt", muon_pt, "pt", 20);
+	 	muonTable->addColumn<float>("eta", muon_eta, "eta", 20);
+	 	muonTable->addColumn<float>("phi", muon_phi, "phi", 20);
+	  muonTable->addColumn<double>("isGlobal", muon_isGlobal, "isGlobal", 20);
+	 	muonTable->addColumn<float>("ptrel", muon_ptrel, "pt relative to parent jet", 20);
+		muonTable->addColumn<int>("nMuHit", muon_nMuHit, "number of muon hits", 20);
+    muonTable->addColumn<int>("nMatched", muon_nMatched, "number of matched stations", 20);
+    muonTable->addColumn<int>("nTkHit", muon_nTkHit, "number of tracker hits", 20);
+    muonTable->addColumn<int>("nPixHit", muon_nPixHit, "number of pixel hits", 20);
+    muonTable->addColumn<int>("nOutHit", muon_nOutHit, "number of missing outer hits", 20);
+    muonTable->addColumn<double>("chi2", muon_chi2, "chi2", 20);
+    muonTable->addColumn<double>("chi2Tk", muon_chi2Tk, "chi2 inner track", 20);
```

Comment:
[Implementation itself, technical details & storage]
The observables of type float or double have precision of 20, while the other variables in JetConstituentTableProducer all use 10. Was there a specific reason to effectively double the precision? Maybe we should stick to one level of precision for jet-constituent-related features.

Reply:
[For my own education, but maybe to avoid we might be missing something here]
How were the stored Muon variables chosen, related to e.g. https://github.com/cms-btv-pog/RecoBTag-PerformanceMeasurements/blob/12_2_X/interface/JetInfoBranches.h#L344-L369?

Reply:
For what concerns the precision, I tried with 10 but I obtained a segmentation violation error. 
About the variables stored, I followed this list according to Soureek instructions [https://github.com/cms-btv-pog/RecoBTag-PerformanceMeasurements/blob/10_6_X_UL2016_PreliminaryJECs/python/varGroups_cfi.py#L1738-L1890](https://github.com/cms-btv-pog/RecoBTag-PerformanceMeasurements/blob/10_6_X_UL2016_PreliminaryJECs/python/varGroups_cfi.py#L1738-L1890)